connect/transforms/src/main/java/org/apache/kafka/connect/transforms/ReplaceField.java

```diff
@@ -94,8 +96,8 @@ public void configure(Map<String, ?> configs) {
             {ConfigName.EXCLUDE, "blacklist"},
         }));
 
-        exclude = config.getList(ConfigName.EXCLUDE);
-        include = config.getList(ConfigName.INCLUDE);
+        exclude = new HashSet<>(config.getList(ConfigName.EXCLUDE));
+        include = new HashSet<>(config.getList(ConfigName.INCLUDE));
```

Comment:
Hm that's an interesting point and I hadn't considered that. However, taking a closer look, I don't think there'll be any performance penalty here because the `ArrayList::contains` method will involve calls to `String::equals` to check whether each value field is present in the exclude / include fields. `String::equals` and `String::hashCode` both have linear time complexity in terms of the length of the String so there won't be much difference in the performance between the `HashSet` version and the `ArrayList` version when there's a small number of include / exclude fields. I ran a modified benchmark and the results were fairly similar between the two implementations:

## Using Strings with length ~10

```
With HashSet:

Benchmark                                                  (fieldCount)  Mode  Cnt       Score      Error  Units
ReplaceFieldBenchmark.includeExcludeReplaceFieldBenchmark           100  avgt    5    1034.411 ±  138.769  ns/op
ReplaceFieldBenchmark.includeExcludeReplaceFieldBenchmark          1000  avgt    5   12210.318 ±  119.902  ns/op
ReplaceFieldBenchmark.includeExcludeReplaceFieldBenchmark         10000  avgt    5  125383.027 ± 3884.894  ns/op

-------------------------------------------------------------------------------------------------------------------

With ArrayList:

Benchmark                                                  (fieldCount)  Mode  Cnt      Score       Error  Units
ReplaceFieldBenchmark.includeExcludeReplaceFieldBenchmark           100  avgt    5    919.979 ±    15.950  ns/op
ReplaceFieldBenchmark.includeExcludeReplaceFieldBenchmark          1000  avgt    5   8757.357 ±   114.363  ns/op
ReplaceFieldBenchmark.includeExcludeReplaceFieldBenchmark         10000  avgt    5  96112.550 ± 10639.302  ns/op
```

## Using Strings with length ~1000

```
With HashSet:

Benchmark                                                  (fieldCount)  Mode  Cnt       Score       Error  Units
ReplaceFieldBenchmark.includeExcludeReplaceFieldBenchmark           100  avgt    5     982.217 ±    16.189  ns/op
ReplaceFieldBenchmark.includeExcludeReplaceFieldBenchmark          1000  avgt    5   12327.410 ±   657.447  ns/op
ReplaceFieldBenchmark.includeExcludeReplaceFieldBenchmark         10000  avgt    5  162847.616 ± 25125.797  ns/op

-------------------------------------------------------------------------------------------------------------------

With ArrayList:

Benchmark                                                  (fieldCount)  Mode  Cnt       Score       Error  Units
ReplaceFieldBenchmark.includeExcludeReplaceFieldBenchmark           100  avgt    5   16189.095 ±   741.592  ns/op
ReplaceFieldBenchmark.includeExcludeReplaceFieldBenchmark          1000  avgt    5   24110.247 ±   180.101  ns/op
ReplaceFieldBenchmark.includeExcludeReplaceFieldBenchmark         10000  avgt    5  165554.095 ± 17875.657  ns/op
```

Even on attempting to make use of String interning to speed up the equals checks, the results were very similar.

Reply:
Only the first call to `String::hashCode` for a string object actually does the hash code computation which is cached and subsequent calls just fetch the cached value.

Reply:
The string interning doesn't make much of a difference since there's only a single field value it benefits (also it'd benefit the `HashSet` implementation as well because of the aforementioned hash code caching).

Admittedly, the benchmark does use strings of identical length for the value fields as well as the include / exclude fields and since the `String::equals` method should likely be O(1) for strings of unequal length, we'd probably see better results for the `ArrayList` based implementation if the benchmark used strings of random length. However, I'd argue that we're probably overthinking it at this point, WDYT? 😅

Reply:
Can't argue with the numbers. Thanks for looking into this!